contracts/AmplificationUtils.sol

```diff
@@ -22,6 +22,65 @@ library AmplificationUtils {
     uint256 private constant MAX_A_CHANGE = 2;
     uint256 private constant MIN_RAMP_TIME = 14 days;
 
+    /**
+     * @notice Return A, the amplification coefficient * n * (n - 1)
+     * @dev See the StableSwap paper for details
+     * @param self Swap struct to read from
+     * @return A parameter
+     */
+    function getA(SwapUtils.Swap storage self) external view returns (uint256) {
+        return _getAPrecise(self).div(A_PRECISION);
+    }
+
+    /**
+     * @notice Return A in its raw precision
+     * @dev See the StableSwap paper for details
+     * @param self Swap struct to read from
+     * @return A parameter in its raw precision form
+     */
+    function getAPrecise(SwapUtils.Swap storage self)
+        external
+        view
+        returns (uint256)
+    {
+        return _getAPrecise(self);
+    }
+
+    /**
+     * @notice Return A in its raw precision
+     * @dev See the StableSwap paper for details
+     * @param self Swap struct to read from
+     * @return A parameter in its raw precision form
+     */
+    function _getAPrecise(SwapUtils.Swap storage self)
```

Comment:
Any reason why you're passing storage here? This function doesn't mutate.

Reply:
Doesn't supplying a storage struct to a memory parameter copy all fields? 
Passing by storage here allows us to read only necessary values inside `self`.

Reply:
Ah yes maybe, you'll need to go through each function and determine if its worth it or not I guess.